tests/ci_folder_parsing.py

```diff
@@ -25,16 +25,24 @@
                        type=str,
                        help="The Class you wish to filter on, (not the Test File)",
                        default=None)
+my_parser.add_argument('-t', "--token",
+                       type=str,
+                       help="The Token associated with the class, such as 'asr1k'",
+                       default=None)
 args = my_parser.parse_args()
 
 _os = args.operating_system
 _class = args.class_name
+_token = args.token
 
 # This is the list of Classes that currently have no testing. It was found during the process
 # of converting to folder based testing strategy
 CLASS_SKIP = {
     "asa": {"ShowVpnSessiondbSuper": True},
     "iosxe": {
+        "c9300": {
+            "ShowInventory": True,
+        },
```

Comment:
I see the issue now, this follow a completely different testing pattern that I had not considered. Can you explain how the mapper functionality works? In the example you have provided, there does not seem to be a human readable way to match there tests up.

Reply:
Because of the name of the class vs the name of the unittest right ? 

The mapper is just a side effect calling a function which is hooked to a dictionary to return the right value

Third example from the top: https://docs.python.org/3/library/unittest.mock.html

Reply:
Let us know if we can help here

Reply:
It would be helpful if you could pull two raw and how they relate, as example, it seems that `https://github.com/CiscoTestAutomation/genieparser/blob/8626c5e3917ac973850bbce5bfe181160720ce68/src/genie/libs/parser/iosxe/tests/test_show_ospf.py#L1560` maps to `https://github.com/CiscoTestAutomation/genieparser/blob/8626c5e3917ac973850bbce5bfe181160720ce68/src/genie/libs/parser/iosxe/tests/test_show_ospf.py#L1006` which I believe led me to think these tests were not working to begin with.

Reply:
Some 30 tests followed this pattern, I am adding back now in their original format. In the future can look to solve this, but for now making sure tests are added back.

The below shows the state before anything was touched (commit `2696b5fac54fc0079a43b1b0e38ffa5636ed340a`) and find the tests in question

```
ntcgenie@e733bff355dc:~/genieparser/tests/asa$ grep -R "def mapper"
ntcgenie@e733bff355dc:~/genieparser/tests/asa$ cd ../ios
ntcgenie@e733bff355dc:~/genieparser/tests/ios$ grep -R "def mapper"
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_protocols.py:        def mapper(key):
test_show_protocols.py:        def mapper(key):
ntcgenie@e733bff355dc:~/genieparser/tests/ios$ cd ../iosxe/
ntcgenie@e733bff355dc:~/genieparser/tests/iosxe$ grep -R "def mapper"
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_ospf.py:        def mapper(key):
test_show_bgp.py:        def mapper(key):
test_show_bgp.py:        def mapper(key):
test_show_bgp.py:        def mapper(key):
test_show_bgp.py:        def mapper(key):
test_show_bgp.py:        def mapper(key):
test_show_protocols.py:        def mapper(key):
test_show_protocols.py:        def mapper(key):
test_show_protocols.py:        def mapper(key):
ntcgenie@e733bff355dc:~/genieparser/tests/iosxe$ git log
commit 2696b5fac54fc0079a43b1b0e38ffa5636ed340a (HEAD -> master)
```